scene/resources/material.cpp

```diff
@@ -1032,6 +1032,8 @@ void BaseMaterial3D::_update_shader() {
 
 	if (features[FEATURE_REFRACTION]) {
 		if (features[FEATURE_NORMAL_MAPPING]) {
+			code += "\tNORMAL_MAP.xy = NORMAL_MAP.xy * 2.0 - 1.0;\n";
```

Comment:
You should be using a local variable such as ``normal_unpacked`` like what Arnklit suggested in the issue. If you override the ``NORMAL_MAP`` variable then the unpack will happen twice for actually calculating the normals of the object.

Reply:
Got it! Yea that actually makes sense... copy the normal mapping, unpack it, and feed it to the shader for when it needs to consider refraction. Leave the original normal mapping unedited for the remainder of the shader.

Something like 
```
normal_unpacked = NORMAL_MAP;
normal_unpacked = ...;
...
code += "\tvec3 ref_normal = normalize( mix(NORMAL,TANGENT * normal_unpacked.x + BINORMAL * normal_unpacked.y + NORMAL * normal_unpacked.z,NORMAL_MAP_DEPTH) );\n";
```

Then? If so I'll get that fixed and ammended to the commit/pr as soon as I get back home! Thanks!